### ISSUE TITLE
ipsec stack stuck - Upgrading state - (Expected state running but got stopped) when trying to upgrade to 0.0.2 version

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#7219]
Rancher server version - Build from master
Add 3 hosts to the setup.
Add catalog - "library" set to -https://github.com/rancher/rancher-catalog and branch v1.3.0-rc2
"Upgrade available" option is available for “ipsec”, “network-service” , "health-check" and scheduler.
Upgraded network-services.
Then attempted to upgrade ipsec stack.
Upgrade is stuck in upgrading state forever with message - ``` Upgrading (Need to restart service reconcile``

<img width="1026" alt="screen shot 2016-12-28 at 7 29 31 pm" src="https://cloud.githubusercontent.com/assets/4266958/21536181/0097b0aa-cd34-11e6-9416-07b6095ed8ee.png">

ipsec-ipsec-router-1 container's log:

```
12/28/2016 7:32:15 PM12[IKE] queueing IKE_INIT task
12/28/2016 7:32:15 PM12[IKE] queueing IKE_NATD task
12/28/2016 7:32:15 PM12[IKE] queueing IKE_CERT_PRE task
12/28/2016 7:32:15 PM12[IKE] queueing IKE_AUTH task
12/28/2016 7:32:15 PM12[IKE] queueing IKE_CERT_POST task
12/28/2016 7:32:15 PM12[IKE] queueing IKE_CONFIG task
12/28/2016 7:32:15 PM12[IKE] queueing IKE_AUTH_LIFETIME task
12/28/2016 7:32:15 PM12[IKE] queueing IKE_MOBIKE task
12/28/2016 7:32:15 PMtime="2016-12-29T03:32:15Z" level=info msg="Loaded connection: conn-35.184.17.111, [aes128gcm16-sha256-modp2048 aes-sha1-modp2048], [aes128gcm16-modp2048 aes-modp2048]"
12/28/2016 7:32:15 PMtime="2016-12-29T03:32:15Z" level=fatal msg="listen tcp :8111: bind: address already in use"
12/28/2016 7:32:15 PM12[I
```

**Comment by deniseschannon** [comment]
Per @ibuildthecloud:

ipsec now is a deployment unit of three services.  ipsec, router, cni-driver.  router is network from ipsec.  router is getting upgraded but the ipsec service is not.  so previously a process ran in ipsec, but it now in router.  router is being launched pointing to the old ipsec container as net:contianer and so there are two processes in the group that are running in the same network namespace.  so router keeps dying, but the basic issue is that router should not be pointing to the old ipsec container

**Comment by alena1108** [comment]
@deniseschannon based on talk with @ibuildthecloud the fix will be on the template side for now. Fixing the upgrade code will require big refactoring that we are trying to avoid at this point

**Comment by deniseschannon** [comment]
please test with rancher/net:v0.8.1, which is with ipsec v0.0.2

**Comment by sangeethah** [comment]
Tested with master builds.
Add 3 hosts to the setup and added few services and LB.
Add catalog - "library" set to -https://github.com/rancher/rancher-catalog and branch v1.3.0-rc3
"Upgrade available" option is available for “ipsec”, “network-service” , "health-check" and scheduler.
Initiated Upgrade on all the 4 services.

Upgrade for ipsec and healthcheck stacks are stuck in "upgrading" state forever.
one of the ipsec container keeps failing to start and gets to “Error” state.

```
mysql> select id,name,state,created,removed from instance where name like "%ipsec%";
+-----+--------------------------+---------+---------------------+---------------------+
| id  | name                     | state   | created             | removed             |
+-----+--------------------------+---------+---------------------+---------------------+
|   2 | ipsec-ipsec-cni-driver-1 | stopped | 2016-12-30 18:02:17 | NULL                |
|   7 | ipsec-ipsec-1            | stopped | 2016-12-30 18:02:17 | NULL                |
|  17 | ipsec-ipsec-cni-driver-2 | stopped | 2016-12-30 18:03:26 | NULL                |
|  19 | ipsec-ipsec-2            | stopped | 2016-12-30 18:03:26 | NULL                |
|  22 | ipsec-ipsec-cni-driver-3 | stopped | 2016-12-30 18:03:27 | NULL                |
|  24 | ipsec-ipsec-3            | stopped | 2016-12-30 18:03:27 | NULL                |
|  43 | ipsec-ipsec-router-1     | purged  | 2016-12-30 18:19:37 | 2016-12-30 18:20:32 |
|  44 | ipsec-ipsec-router-2     | purged  | 2016-12-30 18:19:37 | 2016-12-30 18:19:51 |
|  45 | ipsec-ipsec-router-3     | purged  | 2016-12-30 18:19:38 | 2016-12-30 18:21:00 |
|  46 | ipsec-ipsec-2            | running | 2016-12-30 18:19:49 | NULL                |
|  47 | ipsec-ipsec-router-2     | running | 2016-12-30 18:19:49 | NULL                |
|  49 | ipsec-ipsec-cni-driver-2 | running | 2016-12-30 18:19:50 | NULL                |
|  56 | ipsec-ipsec-1            | running | 2016-12-30 18:20:15 | NULL                |
|  57 | ipsec-ipsec-router-1     | running | 2016-12-30 18:20:15 | NULL                |
|  58 | ipsec-ipsec-cni-driver-1 | running | 2016-12-30 18:20:16 | NULL                |
|  69 | ipsec-ipsec-3            | purged  | 2016-12-30 18:20:59 | 2016-12-30 18:22:28 |
|  70 | ipsec-ipsec-router-3     | purged  | 2016-12-30 18:21:00 | 2016-12-30 18:22:28 |
|  71 | ipsec-ipsec-cni-driver-3 | purged  | 2016-12-30 18:21:00 | 2016-12-30 18:22:29 |
|  85 | ipsec-ipsec-3            | purged  | 2016-12-30 18:22:58 | 2016-12-30 18:24:28 |
|  86 | ipsec-ipsec-router-3     | purged  | 2016-12-30 18:22:58 | 2016-12-30 18:24:28 |
|  87 | ipsec-ipsec-cni-driver-3 | purged  | 2016-12-30 18:22:58 | 2016-12-30 18:24:29 |
|  92 | ipsec-ipsec-3            | purged  | 2016-12-30 18:24:57 | 2016-12-30 18:26:28 |
|  93 | ipsec-ipsec-router-3     | purged  | 2016-12-30 18:24:57 | 2016-12-30 18:26:28 |
|  94 | ipsec-ipsec-cni-driver-3 | purged  | 2016-12-30 18:24:57 | 2016-12-30 18:26:29 |
|  99 | ipsec-ipsec-3            | purged  | 2016-12-30 18:26:58 | 2016-12-30 18:28:29 |
| 100 | ipsec-ipsec-router-3     | purged  | 2016-12-30 18:26:58 | 2016-12-30 18:28:28 |
| 101 | ipsec-ipsec-cni-driver-3 | purged  | 2016-12-30 18:26:58 | 2016-12-30 18:28:29 |
| 106 | ipsec-ipsec-3            | purged  | 2016-12-30 18:28:58 | 2016-12-30 18:30:59 |
| 107 | ipsec-ipsec-router-3     | purged  | 2016-12-30 18:28:58 | 2016-12-30 18:30:59 |
| 108 | ipsec-ipsec-cni-driver-3 | purged  | 2016-12-30 18:28:58 | 2016-12-30 18:31:00 |
| 115 | ipsec-ipsec-3            | purged  | 2016-12-30 18:32:13 | 2016-12-30 18:35:29 |
| 116 | ipsec-ipsec-router-3     | purged  | 2016-12-30 18:32:13 | 2016-12-30 18:35:29 |
| 117 | ipsec-ipsec-cni-driver-3 | purged  | 2016-12-30 18:32:14 | 2016-12-30 18:35:30 |
| 127 | ipsec-ipsec-3            | purged  | 2016-12-30 18:37:44 | 2016-12-30 18:41:00 |
| 128 | ipsec-ipsec-router-3     | purged  | 2016-12-30 18:37:44 | 2016-12-30 18:41:00 |
| 129 | ipsec-ipsec-cni-driver-3 | purged  | 2016-12-30 18:37:44 | 2016-12-30 18:41:01 |
| 139 | ipsec-ipsec-3            | removed | 2016-12-30 18:43:14 | 2016-12-30 18:46:30 |
| 140 | ipsec-ipsec-router-3     | purged  | 2016-12-30 18:43:14 | 2016-12-30 18:46:30 |
| 141 | ipsec-ipsec-cni-driver-3 | removed | 2016-12-30 18:43:14 | 2016-12-30 18:46:31 |
| 150 | ipsec-ipsec-3            | error   | 2016-12-30 18:48:45 | NULL                |
| 151 | ipsec-ipsec-router-3     | stopped | 2016-12-30 18:48:45 | NULL                |
| 152 | ipsec-ipsec-cni-driver-3 | running | 2016-12-30 18:48:45 | NULL                |
+-----+--------------------------+---------+---------------------+---------------------+
42 rows in set (0.00 sec)

```

**Comment by deniseschannon** [comment]
@sangeethah I believe the underlying service reconcile issue in upgraded state is a separate issue, but if you upgrade infra services one by one, then you did not face this problem. 

We should make a new issue for the service reconcile one, and close this issue if the upgrade to this ipsec template succeeded.

**Comment by soumyalj** [comment]
Tested with upgrade from v1.2.2 to v1.3.0-rc6. Each infrastructure stack was upgraded one at a time.  ipsec stack and other stacks upgraded successfully.

**Comment by sangeethah** [comment]
Issue seen when upgrading multiple infrastructure stacks in parallel is tracked in https://github.com/rancher/rancher/issues/7265
